[Issue 157] Support report layout

dalogsdon opened the pull request:
Resolves #157

parente commented:
Hold until jupyter-incubator/dashboards#211 goes in.

parente commented on etc/notebooks/test/test_style_match.ipynb:
```diff
@@ -257,7 +257,7 @@
    },
    "source": [
-    "<img src=\"http://www.bigcat.com/CZ2/cougar2.jpg\">"
+    "<img src=\"http://cincinnatizoo.org/wp-content/uploads/2011/02/cougar1.jpg\">"
```
This was bothering me. Thank you.

parente commented on routes/renderers.js:
```diff
@@ -85,12 +85,12 @@ function _renderDashboard(req, res, next, opts) {
+                                  notebook.metadata.urth &&
+                                  notebook.metadata.urth.dashboard &&
+                                  notebook.metadata.urth.dashboard.layout) ||
+                                  'no-layout';
```
We decided no layout metadata meant render as a report here. But in the extension we said if layout was missing, we'd look for grid metadata and assume grid is the default. Hmmm.

If I take a notebook that was produced by the 0.4.x extension and drop it in dashboard server now with this change, it's going to render in report layout and ignore all the grid metadata. That seems ... unfriendly.

Can we at least do a sniff test to see if any of the cells have the `urth.dashboard` path. If any of them do but layout is undefined in the notebook level metadata, default to grid. If there's no urth.dashboard anywhere then take the no-layout route.

dalogsdon replied:
I agree, and this is what @jameslmartin did in #145. I can reinstate that check.

parente replied:
OK. Cool. Just ping again when it's here and I'll finish testing.

dalogsdon commented:
I updated the renderer to use grid layout if no layout was set but there is at least 1 cell that contains the grid metadata (row, col, width, height). I also noticed the Config script code in the dashboard template was always setting grid layout properties and not always putting in all default values if only some were set in the notebook. I've updated the template to handle these cases.
